### DISCUSSION TITLE
Support for custom parser

### DISCUSSION THREAD

**Discussion by iamprayush** [gorules/zen#149] [Q&A]
Is it possible to use the editor and replace the zen expression language parser with a custom one for specific usecases?

**Comment by stefan-gorules** [comment]
Hi @iamprayush,

You are able to extend GoRules with usage of **custom nodes**. They are currently in beta. Other, more complex option, includes modifying Rust core to suit your needs.

Could you provide a bit more details about what you'd like to achieve? If you can't share this information publicly feel free to reach out at hi@gorules.io.

**Reply by iamprayush** [reply]
Thanks for the suggestion @stefan-gorules. From what I read about the custom nodes, it doesn't seem like they would help in my case.   For my usecase, I want the nodes provided by the editor (decision table and expression), but instead of them validating the zen expression language, I want it to validate a custom expression language using a parser that I can provide.  

I'm essentially trying to figure out how deeply coupled the parser is with the editor and if it's possible to plug in a custom parser.

**Reply by stefan-gorules** [reply]
There's no way of achieving this without modifying the core. The other way is to create a custom node similar to the table that uses your expressions under the hood.

Out of curiosity, are there any limitations that you've faced which have caused you to explore alternative to ZEN expressions?
